bCNC/bmain.py

```diff
@@ -0,0 +1,3030 @@
+        # Program
+        Utils.setInt(Utils.__prg__, "width", str(self.winfo_width()))
+        Utils.setInt(Utils.__prg__, "height", str(self.winfo_height()))
+        Utils.setInt(Utils.__prg__, "sash", str(self.paned.sash_coord(0)[0]))
+
+        # save windowState
+        Utils.setStr(Utils.__prg__, "windowstate", str(self.wm_state()))
+        Utils.setStr(Utils.__prg__, "page", str(
+            self.ribbon.getActivePage().name))
+
+        # Connection
+        Page.saveConfig()
+        Sender.saveConfig(self)
+        self.tools.saveConfig()
+        self.canvasFrame.saveConfig()
+
+    # -----------------------------------------------------------------------
+    def loadHistory(self):
+        try:
+            f = open(Utils.hisFile)
+        except Exception:
+            return
+        self.history = [x.strip() for x in f]
+        self._historySearch = None
+        f.close()
+
+    # -----------------------------------------------------------------------
+    def saveHistory(self):
+        try:
+            f = open(Utils.hisFile, "w")
+        except Exception:
+            return
+        f.write("\n".join(self.history))
+        f.close()
+
+    # -----------------------------------------------------------------------
+    def cut(self, event=None):
+        focus = self.focus_get()
+        if focus in (self.canvas, self.editor):
+            self.editor.cut()
+            return "break"
+
+    # -----------------------------------------------------------------------
+    def copy(self, event=None):
+        focus = self.focus_get()
+        if focus in (self.canvas, self.editor):
+            self.editor.copy()
+            return "break"
+
+    # -----------------------------------------------------------------------
+    def paste(self, event=None):
+        focus = self.focus_get()
+        if focus in (self.canvas, self.editor):
+            self.editor.paste()
+            return "break"
+
+    # -----------------------------------------------------------------------
+    def undo(self, event=None):
+        if not self.running and self.gcode.canUndo():
+            self.gcode.undo()
+            self.editor.fill()
+            self.drawAfter()
+        return "break"
+
+    # -----------------------------------------------------------------------
+    def redo(self, event=None):
+        if not self.running and self.gcode.canRedo():
+            self.gcode.redo()
+            self.editor.fill()
+            self.drawAfter()
+        return "break"
+
+    # -----------------------------------------------------------------------
+    def addUndo(self, undoinfo):
+        self.gcode.addUndo(undoinfo)
+
+    # -----------------------------------------------------------------------
+    def about(self, event=None, timer=None):
+        toplevel = Toplevel(self)
+        toplevel.transient(self)
+        toplevel.title(_("About {} v{}").format(Utils.__prg__, __version__))
+        if sys.platform == "win32":
```

Comment:
Line 829-835 is not needed anymore. This old method actually prevents the correct display of an icon.
Icon bitmap is now set in bmain.py file.
With the following code:
```
        photo = PhotoImage(file = "%s/bCNC.png" % (Utils.prgpath))
        self.iconphoto(True, photo)
```
Before the file split it was enough to only define it once. (as all other windows should inherit the icon from main) Please test if only defining the icon in bmain.py is enough.

Reply:
@m1ch The code is now setting cCNC.png as the icon image.
I think that bCNC.ico and bCNC.xbm can be removed now. Please check.

Reply:
This is resolved in my latest commit.